runtime-modules/forum/src/benchmarking.rs

```diff
@@ -0,0 +1,1110 @@
+
+    }
+    update_category_archival_status{
+        let lead_id = 0;
+
+        let caller_id =
+            insert_a_lead_member::<T>(OpeningType::Leader, lead_id);
+
+        let i in 0 .. T::MaxCategoryDepth::get() as u32;
+
+        let new_archival_status = true;
+
+        // Generate categories tree
+        let (category_id, parent_category_id) = generate_categories_tree::<T>(caller_id.clone(), i);
+
+    }: _ (RawOrigin::Signed(caller_id), PrivilegedActor::Lead, category_id, new_archival_status)
+    verify {
+        let text = vec![0u8].repeat(MAX_BYTES as usize);
+
+        let new_category = Category {
+            title_hash: T::calculate_hash(text.as_slice()),
+            description_hash: T::calculate_hash(text.as_slice()),
+            archived: new_archival_status,
+            num_direct_subcategories: 0,
+            num_direct_threads: 0,
+            num_direct_moderators: 0,
+            parent_category_id,
+            sticky_thread_ids: vec![],
+        };
+
+        assert_eq!(Module::<T>::category_by_id(category_id), new_category);
+        assert_last_event::<T>(RawEvent::CategoryUpdated(category_id, new_archival_status).into());
+    }
+    delete_category {
+
+        let lead_id = 0;
+
+        let caller_id =
+            insert_a_lead_member::<T>(OpeningType::Leader, lead_id);
+
+        let i in 0 .. T::MaxCategoryDepth::get() as u32;
+
+        // Generate categories tree
+        let (category_id, parent_category_id) = generate_categories_tree::<T>(caller_id.clone(), i);
+
+        let category_counter = <Module<T>>::category_counter();
+
+    }: _ (RawOrigin::Signed(caller_id), PrivilegedActor::Lead, category_id)
+    verify {
+        let text = vec![0u8].repeat(MAX_BYTES as usize);
+
+        let new_category: Category<T::CategoryId, T::ThreadId, <T as frame_system::Trait>::Hash> = Category {
+            title_hash: T::calculate_hash(text.as_slice()),
+            description_hash: T::calculate_hash(text.as_slice()),
+            archived: false,
+            num_direct_subcategories: 0,
+            num_direct_threads: 0,
+            num_direct_moderators: 0,
+            parent_category_id: None,
+            sticky_thread_ids: vec![],
+        };
+
+        if let Some(parent_category_id) = parent_category_id {
+            // Ensure number of direct subcategories for parent category decremented successfully
+            assert_eq!(Module::<T>::category_by_id(parent_category_id).num_direct_subcategories, 0);
+        }
+
+        assert_eq!(<Module<T>>::category_counter(), category_counter - T::CategoryId::one());
+
+        // Ensure category removed successfully
+        assert!(!<CategoryById<T>>::contains_key(category_id));
+
+        assert_last_event::<T>(RawEvent::CategoryDeleted(category_id).into());
+    }
+    create_thread {
+        let forum_user_id = 0;
```

Comment:
It seems that you use `forum_user_id` as u32 only once per function benchmark and convert it on other occasions. What do you think about changing it to u64 by default and have only a single conversion to u32 for all benchmarks in the file?

Reply:
Great idea, thank you for the suggestion, https://github.com/Joystream/joystream/pull/1856/commits/dccbf2fb47f4caf123e3a1a35c9dbba01b5c88dc